src/App.tsx

```diff
@@ -1,24 +1,155 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import {
+  addTodo, deleteTodo, getTodos, updateTodos,
+} from './api/todos';
+import { TodoFilter } from './components/TodoFilter';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { ErrorNotification } from './components/ErrorNotification';
+import { Errors } from './types/Errors';
+import { Filters } from './types/Filters';
 
-const USER_ID = 0;
+const USER_ID = 11558;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [errorMesssage, setErrorMesssageState] = useState<string>('');
+  const [filter, setFilter] = useState<Filters>(Filters.All);
+  const [title, setTitle] = useState<string>('');
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [isDisable, setIsDisable] = useState<boolean>(false);
+  const [loadingItems, setLoadingItems] = useState<number[]>([]);
+
+  const setErrorMesssage = (message: string) => {
+    if (message) {
+      setErrorMesssageState(message);
+      setTimeout(() => setErrorMesssageState(''), 3000);
+    } else {
+      setErrorMesssageState('');
+    }
+  };
+
+  const handleTodoClick = (todoId: number) => {
+    const updatedTodos = todos.map((todo) => {
+      if (todo.id === todoId) {
+        return {
+          ...todo,
+          completed: !todo.completed,
+        };
+      }
+
+      return todo;
+    });
+
+    setTodos(updatedTodos);
+    const clickedTodo = todos.find((todo) => todo.id === todoId);
+
+    if (clickedTodo) {
+      updateTodos(clickedTodo);
+    }
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID).then(setTodos)
+      .catch(() => setErrorMesssage(Errors.load));
+  }, []);
```

Comment:
how about loading state for fetching all todos?

you can turn it off on `.finally`

Reply:
+1